contracts/staking/fuse/temp/utils/StakingRewardsPerEpoch.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "./StakingRewards.sol";
+import "./GoodDollarSwaps.sol";
+import "../IConsensus.sol";
+import "../../ISpendingRateOracle.sol";
+import "./ValidatorsManagement.sol";
+
+contract StakingRewardsPerEpoch is StakingRewards {
```

Comment:
i dont see why it should implement stakingrewards
maybe just a mutual interface with some methods and events

Reply:
Well, because the StakingRewardsPerEpoch is an extension of StakingRewards which represents just basic staking

Reply:
And so we don't have to duplicate all the function validation modifiers (like nonReentrant and other stuff)